src/scripts/generate_relation_and_equality_ops.py

```diff
@@ -0,0 +1,280 @@
+              "less": self.less,
+              "more": self.more,
+              "lessorequ": self.lessorequ,
+              "moreorequ": self.moreorequ,
+              "equ": self.equ,
+              "sequ": self.sequ,
+              "nequ": self.nequ,
+              "snequ": self.snequ,
+        } # Functions for the operators
+
+    # Generating the expression with operands and operators
+    def expression_maker(self):
+        for op in self.options.test_cases:
+            self.operators_list.append(self.oper_funcs[op])
+
+        self.operands = [0] * self.num_operands # Creating a list for the operands
+        self.operators = [0] * (self.num_operands - 1) # Creating a list for the operators
+
+        # Generating the expression
+        for i in range(self.num_operands - 1):
+            decider = random.randint(0, 10)
+            if (decider == 0):
+                self.operands[i] = "undefined"
+            elif (decider == 1):
+                self.operands[i] = "null"
+            elif (decider == 2):
+                self.operands[i] = "true"
+            elif (decider == 3):
+                self.operands[i] = "false"
+            elif (decider == 4):
+                self.operands[i] = "NaN"
+            elif (decider == 5):
+                self.operands[i] = str(random.randint(_constants.min, _constants.max))
+            else:
+                self.operands[i] = random.randint(_constants.min, _constants.max)
+            self.operators[i] = random.randint(0, len(self.operators_list) - 1)
+        self.operands[self.num_operands - 1] = random.randint(_constants.min, _constants.max)
+
+    # Convert python boolean values to fit js
+    def convert_true_false_to_lowercase_string(self, value):
+        if value:
+            converted = "true"
+        else:
+            converted = "false"
+        return converted
+
+    # Add comas to the value if it is meant to be a string in the test case
+    def add_coma_if_string(self, value):
+        if value not in self._InnerValues.special_values and isinstance(value, str):
+            value = "'%s'" % (value)
+        return value
+
+    # ES v5.1 11.8. - 11.9. (except 11.8.7.)
+    # Relational and equality operators
+
+    # Equality operators
+
+    # ES v5.1 11.9.3
+    # Equality
+    def equ(self, left_value, right_value):
+        # If one of the values is NaN return false
+        if (left_value == "NaN" or right_value == "NaN"):
+            return GenSequence.create_operation_result_object(self ,'==', right_value, "false")
+
+        # Cases where it should return true
+        elif ((left_value == "undefined" and right_value == "undefined") or
+        (left_value == "null" and right_value == "null") or
+        (left_value == "undefined" and right_value == "null") or
+        (left_value == "null" and right_value == "undefined") or
+        (left_value == "true" and right_value == "true") or
+        (left_value == "false" and right_value == "false")):
```

Comment:
Create a mapping or something for these special cases. This look like a but ugly.